akka-actor/src/main/scala/akka/actor/AbstractActor.scala

```diff
@@ -29,57 +86,138 @@ object AbstractActor {
-      "use context().become(...) to change it later")
+  def getSelf(): ActorRef = self
 
   /**
-   * Returns this AbstractActor's AbstractActorContext
-   * The AbstractActorContext is not thread safe so do not expose it outside of the
-   * AbstractActor.
+   * The reference sender Actor of the currently processed message. This is
+   * always a legal destination to send to, even if there is no logical recipient
+   * for the reply, in which case it will be sent to the dead letter mailbox.
+   *
+   * Same as `sender()`.
+   *
+   * WARNING: Only valid within the Actor itself, so do not close over it and
+   * publish it to other threads!
+   */
+  def getSender(): ActorRef = sender()
+
+  /**
+   * User overridable definition the strategy to use for supervising
+   * child actors.
+   */
+  override def supervisorStrategy: SupervisorStrategy = super.supervisorStrategy
+
+  /**
+   * User overridable callback.
+   * <p/>
+   * Is called when an Actor is started.
+   * Actor are automatically started asynchronously when created.
+   * Empty default implementation.
    */
-  def getContext(): AbstractActorContext = context.asInstanceOf[AbstractActorContext]
+  @throws(classOf[Exception])
+  override def preStart(): Unit = super.preStart()
 
-  override def receive =
-    if (_receive != null) _receive
-    else throw IllegalActorStateException("Actor behavior has not been set with receive(...)")
+  /**
+   * User overridable callback.
+   * <p/>
+   * Is called asynchronously after `getContext().stop()` is invoked.
+   * Empty default implementation.
+   */
+  @throws(classOf[Exception])
+  override def postStop(): Unit = super.postStop()
+
+  /**
+   * User overridable callback: '''By default it disposes of all children and then calls `postStop()`.'''
+   * <p/>
+   * Is called on a crashed Actor right BEFORE it is restarted to allow clean
+   * up of resources before Actor is terminated.
+   */
+  @throws(classOf[Exception])
+  override def preRestart(reason: Throwable, message: Option[Any]): Unit = super.preRestart(reason, message)
+
+  /**
+   * User overridable callback: By default it calls `preStart()`.
+   * <p/>
+   * Is called right AFTER restart on the newly created Actor to allow reinitialization after an Actor crash.
+   */
+  @throws(classOf[Exception])
+  override def postRestart(reason: Throwable): Unit = super.postRestart(reason)
+
+  /**
+   * An actor has to define its initial receive behavior by implementing
+   * the `initialReceive` method.
+   */
+  def initialReceive(): AbstractActor.Receive
+
+  override def receive: PartialFunction[Any, Unit] =
+    initialReceive().onMessage.asInstanceOf[PartialFunction[Any, Unit]]
+
+  /**
+   * Convenience factory of the `ReceiveBuilder`.
+   * Creates a new empty `ReceiveBuilder`.
+   */
+  final def receiveBuilder(): ReceiveBuilder = ReceiveBuilder.create()
+}
+
+abstract class UntypedAbstractActor extends AbstractActor {
```

Comment:
Doc on the class to highlight why it exists and pointing out that directly implementing `AbstractActor` is better?